src/liballoc/lib.rs

```diff
@@ -69,7 +69,10 @@
        issue_tracker_base_url = "https://github.com/rust-lang/rust/issues/",
        test(no_crate_inject, attr(allow(unused_variables), deny(warnings))))]
 #![no_std]
-#![needs_allocator]
+
+// Don't inject an allocator in stage0 since we explicitly use alloc_system, jemalloc
+// being possibly unavailable or disallowed
+#![cfg_attr(not(stage0), needs_allocator)]
```

Comment:
Honestly, I picked the second issue because it was comitted together with the first one. Furthermore, I am not sure when `make install` should not start at stage0. If I do a change in Rust, `make install` will always start at stage0, right?